src/streaming/TextSourceBuffer.js

```diff
@@ -29,56 +29,103 @@
+                this.buffered =  this.system.getObject("customTimeRanges");
+                this.initializationSegmentReceived= false;
+                this.timescale= 90000;
+                this.allTracks = this.mediaController.getTracksFor("fragmentedText", this.streamController.getActiveStreamInfo());
+            }
         },
+
         append: function (bytes, chunk) {
             var self = this,
                 result,
-                label,
-                lang,
                 samplesInfo,
                 i,
-                ccContent;
+                ccContent,
+                mediaInfo = chunk.mediaInfo,
+                mimeType = mediaInfo.type === "fragmentedText" ? mediaInfo.type : mediaInfo.mimeType;
+
+            function createTextTrackFromMediaInfo(captionData, mediaInfo) {
+                var textTrackInfo = new MediaPlayer.vo.TextTrackInfo();
+                textTrackInfo.captionData = captionData;
+                textTrackInfo.lang = mediaInfo.lang;
+                textTrackInfo.label = mediaInfo.id;
+                textTrackInfo.video = self.videoModel.getElement();
+                textTrackInfo.defaultTrack = self.getIsDefault(mediaInfo);
+                textTrackInfo.isFragmented = self.isFragmented;
+                textTrackInfo.role = (mediaInfo.roles.length > 0) ? mediaInfo.roles[0] : null;
+                self.textTrackExtensions.addTextTrack(textTrackInfo, self.mediaInfos.length);
+                self.eventBus.dispatchEvent({type:MediaPlayer.events.TEXT_TRACK_ADDED});
+            }
 
             if(mimeType=="fragmentedText"){
-                var fragmentExt;
+                var fragmentExt = self.system.getObject("fragmentExt");
                 if(!this.initializationSegmentReceived){
                     this.initializationSegmentReceived=true;
-                    label = mediaInfo.id;
-                    lang = mediaInfo.lang;
-                    this.textTrackExtensions=self.getTextTrackExtensions();
-                    this.textTrackExtensions.addTextTrack(self.videoModel.getElement(), result, label, lang, true);
-                    self.eventBus.dispatchEvent({type:MediaPlayer.events.TEXT_TRACK_ADDED});
-                    fragmentExt = self.system.getObject("fragmentExt");
-                    this.timescale= fragmentExt.getMediaTimescaleFromMoov(bytes);
-                }else{
-                    fragmentExt = self.system.getObject("fragmentExt");
-
-                    samplesInfo=fragmentExt.getSamplesInfo(bytes);
-                    for(i= 0 ; i<samplesInfo.length ;i++) {
+                    for (i = 0; i < this.mediaInfos.length; i++){
+                        createTextTrackFromMediaInfo(null, this.mediaInfos[i]);
+                    }
+                    self.videoModel.getElement().textTracks.addEventListener('change', onTextTrackChange.bind(self));
+                    this.timescale = fragmentExt.getMediaTimescaleFromMoov(bytes);
```

Comment:
I believe the timescale always returns NaN unless it is the initial segment in which I extract the value only once at that time.   Not sure what a better way to do this would be.  Any suggestions.  

Also talking to EBU about EBU-TT it seems that we should use DTS not CTS for the start times.  Any opinion on this topic?